src/policy/immix/immixspace.rs

```diff
@@ -592,15 +592,12 @@ impl<Edges: ProcessEdgesWork> ScanObjectsAndMarkLines<Edges> {
 }
 
 impl<E: ProcessEdgesWork> GCWork<E::VM> for ScanObjectsAndMarkLines<E> {
-    fn do_work(&mut self, worker: &mut GCWorker<E::VM>, mmtk: &'static MMTK<E::VM>) {
+    fn do_work(&mut self, worker: &mut GCWorker<E::VM>, _mmtk: &'static MMTK<E::VM>) {
```

Comment:
As `GCWorker` has a field that points to `MMTK`, it seems we do not need this `mmtk` argument in `do_work()`. Right? No change is needed for the PR.

Reply:
That's right.  We don't need the `&MMTK<VM>` argument to `do_work`, because `GCWorker` already refers to all shared and owned data structures it needs for executing `GCWork`.  Not only that, but `GCController` also has an `&MMTK<VM>` reference, because it is a thread-local context, too, and `mmtk` is something it needs for doing its work.

I made `GCWorker::mmtk` public. I think it should be accessible to `GCWork`, because `GCWork`, by definition, is a piece of work `GCWorker` performs, and should have access to `GCWorker` fields.